### ISSUE TITLE
Include pull request template file in pull request message

### ISSUE THREAD

**Issue by tobiasweibel** [renovatebot/renovate#17897]
### What would you like Renovate to be able to do?

As explained in https://github.com/renovatebot/renovate/discussions/17888 it would be good if the pull request template of the repository can be added to the pull request message. It will help to comply with some pull request requirements of some repositories / companies.

### If you have any ideas on how this should be implemented, please tell us here.

As an example for this repository one could modify the [./renovate.json](https://github.com/renovatebot/renovate/blob/main/renovate.json) (or [../.github/default.json](https://github.com/renovatebot/.github/blob/main/default.json)) and add `prFooter` (or `prHeader`, ...) to include [../renovatebot/.github/pull_request_template.md](https://github.com/renovatebot/renovate/blob/main/.github/pull_request_template.md).
Maybe it would be better to have an explicit option to enable that feature. Options could be `true`, `false`, or a path to a file (e.g. `github>renovatebot/.github/pull_request_template.md`).

A simpler approach would be to just include the pull request template if there is such a file at the supported folders. See [GitHub documentation](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository). Obviously different implementations are needed for other supported platforms (gitlab, ...).

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
Although it would be nice to do this automatically, here are the reasons why not to:
- Even if users have PR templates, they may not need/want them used by Renovate. This very repo would be an example of that
- If users do want the template used, we can't be sure if they want it as header or footer
- GitHub's logic for PR templates may change regularly (e.g. repo and file naming conventions)

So therefore I support the idea to make this explicitly configured instead of Renovate needing to guess or derive it from multiple sources.

The next question is what syntax to support. It seems you're reusing the syntax we have for presets, although instead of returning a parsed result in JSON format your expectation would be a raw file content. I think our platforms each support `getRawFile()` so this is probably OK. I wonder if we should limit it to `local>` though which would cover most of the use cases anyway.

**Comment by tobiasweibel** [comment]
I also prefer the explicit configuration.

`local>` is probably enough, I think (or guess) that only local pr templates are supported by GitHub.

Syntax: I don't know Renovate config good enough to really give an answer. Do you think
```json
{
  "prHeader": "local>.github/pull_request_template.md",
  "prFooter": "local>.github/pull_request_template.md",
}
```
would be OK (only one of prHeader/prFooter really make sense)?
What if users try to combine it with other strings. E.g. `"local>.github/pull_request_template.md\n\nThis PR has been generated by [Renovate Bot](https://github.com/renovatebot/renovate)."`. Should that be supported?

**Comment by rarkins** [comment]
I don't think we need to restrict it to the same org, so therefore enforcing org/repo in the string would be important.

**Comment by tobiasweibel** [comment]
@rarkins I'm out for a week, happy to discuss afterwards.

**Comment by rarkins** [comment]
Let's use the same syntax/convention as for presets - similar to how you proposed - however enforce that it must be a full filename at end (unlikely presets which have an implicit `default.json` if filename is unspecified).

I would like to leverage our existing templating+compiling syntax in the PR body itself, e.g. `{{ something }}`. We can either:
- Hack/overrride this to support our own syntax, and make sure to replace it prior to Handlebars compilation, or
- Make each imported file be defined in some type of "importedFiles" object in config, and then it can be along the lines of `{{ imports.prTemplate }}` and work with Handlebars

The latter would make for easier validation. Something like this:

```
{
  "importedPrBodyFiles": {
    "prTemplate": "local>abc/def:.github/pull_request_template.md"
  },
  "prHeader": "{{ imports.prTemplate }}"
}
```

@viceice @JamieMagee any thoughts?

**Comment by viceice** [comment]
latter sounds good to me

**Comment by realloser** [comment]
## Header or Footer

Having an option in the configuration will probably wipe away current text when a PR is updated by RenovateBot (my assumption on the implementation). In other words, existing changes to the body will be lost (ticket check boxes, comments). Wouldn't it be easier to have a place holder in the template itself instead of header/footer in the configuration and also use it for all other RenovateBot PR. With that, one can specifically choose where to place the RenovateBot specific body and RenovateBot would always only replace this section. Obviously it would still need a configuration to choose what template to use if there are more than one available. Making an arbitrary example:

``` MD
<!-- PULL_REQUEST_TEMPLATE.md -->

## Description:

...

## Checklist: 

...

## Other: 

<!-- RenovateBot start -->
RenovateBot replaces this part with every PR and update of a PR rest remains untouched
<!-- RenovateBot end -->

## Conclusion:

...
```

**Comment by DevHazzardt** [comment]
Both these changes would fit our needs perfectly, is this still being assessed or is this a feature that will be implemented?

**Comment by rarkins** [comment]
#18915 is marked as status:ready, and is probably the best idea to start with. PRs welcome
